etna/metrics/functional_metrics.py

```diff
@@ -41,6 +42,52 @@ def _get_axis_by_multioutput(multioutput: str) -> Optional[int]:
         assert_never(multioutput_enum)
 
 
+def mse_with_missing_handling(y_true: ArrayLike, y_pred: ArrayLike, multioutput: str = "joint") -> ArrayLike:
+    """Mean squared error with missing values handling.
+
+    `Wikipedia entry on the Mean squared error
+    <https://en.wikipedia.org/wiki/Mean_squared_error>`_
+
+    The nans are ignored during computation.
```

Comment:
It would be helpful to also note what will be returned if all nans in the segment